test/tools/Modules/HelpersDebugger/HelpersDebugger.psm1

```
@@ -0,0 +1,220 @@
+        [Alias('sb')]
+        [ScriptBlock]
+        $ScriptBlock,
+
+        [Parameter()]
+        [ValidateNotNullOrEmpty()]
+        [string[]]
+        $CommandQueue
+    )
+    try {
+        $callerEAP = $ErrorActionPreference
+        #  If the debugger is not set up properly, notify the user with an error message
+        if (-not $script:debuggerStopHandlerRegistered -or $host.DebuggerEnabled) {
+            $message = 'You must invoke Register-DebuggerHandler before invoking Test-Debugger, and Unregister-DebuggerHandler after invoking Test-Debugger. As a best practice, invoke Register-DebuggerHandler in the BeforeAll block and Unregister-DebuggerHandler in the AfterAll block of your test script.'
+            $exception = New-Object -TypeName System.InvalidOperationException -ArgumentList $message
+            $errorRecord = New-Object -TypeName System.Management.Automation.ErrorRecord -ArgumentList $exception,$exception.GetType().Name,'InvalidOperation'
+            throw $errorRecord
+        }
+        $script:dbgResults = @()
+        $script:dbgCmdQueue = [System.Collections.Queue]::new()
+        foreach ($command in $CommandQueue) {
+            $script:dbgCmdQueue.Enqueue($command)
+        }
+        # We re-create the script block before invoking it to ensure that it will
+        # work regardless of where the script itself was defined in the test file.
+        # We also suppress any exceptions and silence any output because this
+        # invocation is about the debugger output, not the output of the script
+        # itself
+        & {
+            [System.Diagnostics.DebuggerStepThrough()]
+            [CmdletBinding()]
+            param()
+            try {
+                $ErrorActionPreference = [System.Management.Automation.ActionPreference]::Stop
+                [ScriptBlock]::Create($ScriptBlock).Invoke() > $null
+            } catch {
+                Write-Error -ErrorRecord $_ -ErrorAction Stop
```

> We also suppress any exceptions and silence any output because this ...

If the aim is to suppress any exceptions, maybe we should not write out error in this case?